### PR TITLE
Change MD5 calls

### PR DESCRIPTION
**Pull request by nimrod-becker**
### Explain the changes
1. Agent Wrapper remove md5 validation 
2. SensitiveString change MD5 for printing to Sha258 truncated to 128.

### CODE REVIEW

**Review by jackyalbo — APPROVED** [review]

File: src/util/sensitive_string.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ class SensitiveString {
             this.val = val.unwrap();
         } else if (type === 'string') {
             this.val = val;
-            this.md5 = crypto.createHash('md5').update(this.val).digest('hex');
+            const sha = crypto.createHash('sha256').update(this.val).digest('hex');
+            this.md5 = sha.slice(0, sha.length / 2);
```
Comment (liranmauda) [inline src/util/sensitive_string.js:16]: shouldn't we call it something else if we are not using md5?

**Review by liranmauda — APPROVED** [review]
LGTM

**Review by jackyalbo — APPROVED** [review]

File: src/util/sensitive_string.js [hunk 2]
```diff
@@ -8,21 +8,22 @@ class SensitiveString {
     constructor(val) {
         const type = typeof val;
         if (val instanceof SensitiveString) {
-            this.md5 = val.md5;
+            this.hash = val.hash;
             this.val = val.unwrap();
         } else if (type === 'string') {
             this.val = val;
-            this.md5 = crypto.createHash('md5').update(this.val).digest('hex');
+            const sha = crypto.createHash('sha256').update(this.val).digest('hex');
```
Comment (guymguym) [inline src/util/sensitive_string.js:15]: sha512 is still faster that sha256 so I would use it

File: src/util/sensitive_string.js [hunk 3]
```diff
@@ -8,21 +8,22 @@ class SensitiveString {
...
+            this.hash = sha.slice(0, sha.length / 2);
```
Comment (guymguym) [inline src/util/sensitive_string.js:16]: I would actually slice a fixed size - say first 16 hex digits (64bits).

**Review by guymguym — APPROVED** [review]
See minor comments

**Pull request merged**